pkg/nsx/services/subnetport/builder.go

```diff
@@ -184,11 +184,16 @@ func buildSubnetPortExternalAddressBindingFromExisting(subnetPort *model.VpcSubn
 	return subnetPort
 }
 
-func (service *SubnetPortService) buildExternalAddressBinding(sp *v1alpha1.SubnetPort) *model.ExternalAddressBinding {
-	if service.GetAddressBindingBySubnetPort(sp) != nil {
-		return &model.ExternalAddressBinding{}
+func (service *SubnetPortService) buildExternalAddressBinding(sp *v1alpha1.SubnetPort, restoreMode bool) *model.ExternalAddressBinding {
+	addressBinding := service.GetAddressBindingBySubnetPort(sp)
+	if addressBinding == nil {
+		return nil
 	}
-	return nil
+	portExternalAddressBinding := &model.ExternalAddressBinding{}
+	if restoreMode && len(addressBinding.Status.IPAddress) > 0 {
+		portExternalAddressBinding.ExternalIpAddress = String(addressBinding.Status.IPAddress)
```

Comment:
Do you allocate IPAddressAllocation for the external address binding in restore mode

Reply:
Thanks for the reminder! Added.